test-suite/handwritten-src/objc/tests/DBAsyncTest.m

```diff
@@ -0,0 +1,79 @@
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+#import <DJFuture.h>
+#import "DBTestHelpers.h"
+#import "DBAsyncInterface.h"
+
+@interface AsyncInterfaceImpl : NSObject <DBAsyncInterface>
+@end
+
+@implementation AsyncInterfaceImpl
+- (DJFuture<NSString *> *)futureRoundtrip:(DJFuture<NSNumber *> *)f {
+    return [f then:^id(DJFuture<NSNumber*>* i) { return [[i get] stringValue]; }];
+}
+@end
+
+@interface DBAsyncTests : XCTestCase
+@end
+
+@implementation DBAsyncTests
+
+- (void)setUp
+{
+    [super setUp];
+}
+
+- (void)tearDown
+{
+    [super tearDown];
+}
+
+- (void)testConsumeNativeFuture {
+    DJFuture<NSNumber*>* f1 = [DBTestHelpers getAsyncResult];
+    DJFuture<NSString*>* f2 =[f1 then:^id(DJFuture<NSNumber*>* res) {
+            return [[res get] stringValue];
+        }];
+    DJFuture<NSNumber*>* f3 = [f2 then:^id(DJFuture<NSString*>* res) {
+            return @([[res get] integerValue]);
+        }];
+    NSNumber* i = [f3 get];
+    XCTAssertEqual([i integerValue], 42);
+}
+
+- (void) testFutureRoundtrip {
+    DJPromise<NSString*>* p = [[DJPromise alloc] init];
+    DJFuture<NSString*>* f = [p getFuture];
+    DJFuture<NSNumber*>* f2 = [f then:^id(DJFuture<NSString*>* s) {
+            return @([[s get] integerValue]);
+        }];
+    DJFuture<NSString*>* f3 = [DBTestHelpers futureRoundtrip:f2];
+    [p setValue:@"36"];
+    XCTAssertEqualObjects([f3 get], @"36");
+}
+
+- (void) testFutureRoundtripWithException {
+    DJPromise<NSString*>* p = [[DJPromise alloc] init];
+    DJFuture<NSString*>* f = [p getFuture];
+    DJFuture<NSNumber*>* f2 = [f then:^id(DJFuture<NSString*>* s) {
+            if (1) {
```

Comment:
`if (1)` ?

Reply:
Yes I want to always throw in this function, but without this "if", the compile would complain about either return type or dead code.

Reply:
It would be nice to have the above as a comment in the code.
This test file is also a showcase of this new addition to djinni.